### PR TITLE
Create eip-4573.md

### PR DESCRIPTION
**Pull request by gcolvin**

### CODE REVIEW

File: EIPS/eip-4573.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+         . . .
+         outputs
+         . . .
+```
+
+*Note:  In order to save space the inputs and outputs overlap.  When the inputs are used up in the course of producing the outputs this should not be a problem.  (The space beyond the end of the call frame can be used for temporaries.)*
+
+### Instructions
+
+#### ENTERPROC (0x??) n_inputs: uint2, n_locals: uint2, n_outputs: uint2
```
Comment (shemnon) [inline EIPS/eip-4573.md:75]: s/uint2/uint256/?  Surely you expect more than 3 inputs, outputs and locals.

Comment (gcolvin) [reply EIPS/eip-4573.md:75]: I better  add more description ;)  Those  are the counts of how many inputs, locals, and outputs are allocated.   The way I use them elsewhere is consistent with this.

**Comment by pepyakin** [comment]
Did you consider leveraging a varint instead of a 16-bit unsigned? Common case would be more compact whereas bigger numbers expressible.

**Comment by gcolvin** [comment]
> Did you consider leveraging a varint instead of a 16-bit unsigned? Common case would be more compact whereas bigger numbers expressible.

Bigger numbers aren't really needed, (65,565 parameters?!) and slowing down decoding to save a little space didn't seem worth it -- thus  EIP-4200 when with 2 bytes.  But even 127 parameters is probably enough, so we could go with one byte for now and leave a bit for expansion.  Worth discussing on the Magician's thread.

**Comment by pepyakin** [comment]
> But even 127 parameters is probably enough, so we could go with one byte for now and leave a bit for expansion

Agreed

**Comment by gcolvin** [comment]
Bot alternated complaining about spelling and trailing whitespace, but seems to only complain about trailing white space now.  All I can see is the very last empty line, which the Github editor puts back if I remove it.

File: EIPS/eip-4573.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
+---
+eip: 4573
+title: Entry Points and Procedures for EVM Code Sections
+description: This proposal introduces EVM procedures as entry points to EOF code sections.
+status: Draft
+type: Standards Track 
+category: Core
+author: Greg Colvin <greg@colvin.org>
```
Comment (MicahZoltu) [inline EIPS/eip-4573.md:8]: At least one author should have GitHub contact info so the bot can work properly.  Either of the following would be fine in this case (author's choice).
```suggestion
author: Greg Colvin (@gcolvin), Greg Colvin<greg@colvin.org>
```
```suggestion
author: Greg Colvin (@gcolvin)
```

File: EIPS/eip-4573.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+## Backwards Compatibility
+
+This proposal adds new EOF sections and EVM opcodes.  It doesn't remove or change the semantics of any existing opcodes, so there should be no backwards compatibility issues.
+
+## Security
+
+Safe use of these constructs will be checked completely at validation time, in linear time and space, so there should be no security issues.
+
+
```
Comment (MicahZoltu) [inline EIPS/eip-4573.md:145]:
```suggestion
```

File: EIPS/eip-4573.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+We propose to treat memory addresses as signed, so the formula for `MSTOREFP` needs to use an absolute value
+>$μ′_{i} ≡ max(abs(μ_{i}),⌈(μ_{s}[0]+32)÷32⌉)$
+
+*TBD: Details!*
+
+## Rationale
+
+This proposal uses the [EIP-2315](./eip-2315) return stack to manage calls and returns, and steals and simplifies ideas from [EIP-615](./eip-615), [EIP-3336](./eip-3336) and [EIP-3337](./eip-3337). `ENTERPROC` is modeled on `BEGINSUB`, and per EIP-3336 and EIP-3337 it moves call frames from the data stack to a new in-memory stack. The `FP` register automatically tracks call-frame addresses as procedures are entered and left, and cannot be otherwise modified
+*TBD: The aliasing of the frame stack with ordinary memory supports languages that provide pointers to variables on the stack
```
Comment (MicahZoltu) [inline EIPS/eip-4573.md:135]:
```suggestion
 *TBD: The aliasing of the frame stack with ordinary memory supports languages that provide pointers to variables on the stack*
```

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
Only real blocker is the author line.  A couple of formatting notes included as well.

**Review by MicahZoltu — APPROVED** [review]

File: EIPS/eip-4573.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
+---
+eip: 4573
+title: Entry Points and Procedures for EVM Code Sections
+description: This proposal introduces EVM procedures as entry points to EOF code sections.
+status: Draft
+type: Standards Track
```
Comment (MicahZoltu) [inline EIPS/eip-4573.md:6]:
```suggestion
type: Standards Track
```
This should satisfy the bot.  There is an issue in the backlog to improve the error messaging around this: https://github.com/lightclient/eipv/issues/20

**Comment by gcolvin** [comment]
Thanks @MicahZoltu

**Pull request merged**